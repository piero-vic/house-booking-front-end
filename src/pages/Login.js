/* eslint-disable react/jsx-props-no-spreading */
import { useState } from 'react';
import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import { Link, Navigate } from 'react-router-dom';
import { loginUser } from '../redux/actions/auth';

const Login = ({ loggedIn }) => {
  if (loggedIn) return <Navigate to="/" replace />;

  const dispatch = useDispatch();
  const [error, setError] = useState(null);
  const { register, handleSubmit, formState: { errors } } = useForm();

  const onFormSubmit = (data) => dispatch(loginUser(data)).catch(() => setError('Invalid credentials. Try again'));

  return (
    <main className="min-h-screen min-w-full bg-center bg-cover bg-no-repeat bg-login">
      <div className="min-h-screen min-w-full grid place-items-center bg-lime-500 bg-opacity-80">
        <div className="w-3/4 md:w-1/2">
          <div className="border-b border-b-white p-5 mb-5 mx-auto w-10/12">
            <h2 className="text-white text-center font-bold text-4xl">LOG IN</h2>
          </div>
          {error && <p className="text-red-700 font-bold mb-1 text-center">{error}</p>}
          {errors.password && <p className="text-red-700 font-bold mb-1 text-center">Password is required</p>}
          {errors.email && <p className="text-red-700 font-bold mb-1 text-center">Email is required</p>}
          <form className="flex flex-col gap-5 items-center" onSubmit={handleSubmit(onFormSubmit)}>
            <div className="w-full">
              <input className="w-full bg-white py-2 px-4 rounded-3xl" type="email" placeholder="Email" {...register('email', { required: 'Email is required' })} />
            </div>
            <div className="w-full">
              <input className="w-full bg-white py-2 px-4 rounded-3xl" type="password" placeholder="Password" {...register('password', { required: 'Password is required' })} />
            </div>
            <input className="w-1/2 bg-lime-500 text-white py-2 px-4 border-2 border-white rounded-3xl font-bold" type="submit" value="Log In" />
            <Link className="w-1/2 bg-white text-lime-500 py-2 px-4 border-2 border-lime-500 rounded-3xl font-bold text-center" to="/signup">Sign Up</Link>
          </form>
        </div>
      </div>
    </main>
  );
};

Login.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
};

export default Login;
