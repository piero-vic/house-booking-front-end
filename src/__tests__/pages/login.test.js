import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Login from '../../pages/Login';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Login loggedIn={false} />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
