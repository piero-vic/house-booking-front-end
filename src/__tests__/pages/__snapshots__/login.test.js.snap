// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="min-h-screen min-w-full bg-center bg-cover bg-no-repeat bg-login"
>
  <div
    className="min-h-screen min-w-full grid place-items-center bg-lime-500 bg-opacity-80"
  >
    <div
      className="w-3/4 md:w-1/2"
    >
      <div
        className="border-b border-b-white p-5 mb-5 mx-auto w-10/12"
      >
        <h2
          className="text-white text-center font-bold text-4xl"
        >
          LOG IN
        </h2>
      </div>
      <form
        className="flex flex-col gap-5 items-center"
        onSubmit={[Function]}
      >
        <div
          className="w-full"
        >
          <input
            className="w-full bg-white py-2 px-4 rounded-3xl"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            type="email"
          />
        </div>
        <div
          className="w-full"
        >
          <input
            className="w-full bg-white py-2 px-4 rounded-3xl"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
        </div>
        <input
          className="w-1/2 bg-lime-500 text-white py-2 px-4 border-2 border-white rounded-3xl font-bold"
          type="submit"
          value="Log In"
        />
        <a
          className="w-1/2 bg-white text-lime-500 py-2 px-4 border-2 border-lime-500 rounded-3xl font-bold text-center"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </form>
    </div>
  </div>
</main>
`;
